h2o-algos/src/main/java/hex/glrm/GLRMGenX.java

```diff
@@ -0,0 +1,95 @@
+package hex.glrm;
+
+import hex.genmodel.GenModel;
+import hex.genmodel.MojoModel;
+import hex.genmodel.algos.glrm.GlrmMojoModel;
+import hex.genmodel.easy.EasyPredictModelWrapper;
+import hex.genmodel.easy.RowData;
+import hex.genmodel.easy.exception.PredictException;
+import hex.genmodel.easy.prediction.AbstractPrediction;
+import hex.genmodel.easy.prediction.DimReductionModelPrediction;
+import water.H2O;
+import water.MRTask;
+import water.api.StreamingSchema;
+import water.fvec.Chunk;
+import water.fvec.NewChunk;
+import water.util.JCodeGen;
+import water.util.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+/**
+ * GLRMGenX will generate the coefficients (X matrix) of a GLRM model given the archetype
+ * for a dataframe.
+ */
+public class GLRMGenX  extends MRTask<GLRMGenX> {
+  final GLRMModel _m;
+  final int _k;   // store column size of X matrix
+
+
+  public GLRMGenX(GLRMModel m, int k) {
+    _m = m;
+    _m._parms = m._parms;
+    _k = k;
+
+  }
+
+  public GenModel generateMojo(int cidx) {
+    String modelName = JCodeGen.toJavaId(_m._key.toString());
```

Comment:
We should avoid writing the mojo file for every chunk.

Would it be possible to refactor the code so that we can use the score0 function of GlrmMojoModel directly?